src/pipeline/src/etl/processor/gsub.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use regex::Regex;
+
+use crate::etl::field::{Field, Fields};
+use crate::etl::processor::{
+    yaml_bool, yaml_field, yaml_fields, yaml_string, Processor, FIELDS_NAME, FIELD_NAME,
+    IGNORE_MISSING_NAME,
+};
+use crate::etl::value::{Array, Map, Value};
+
+pub(crate) const PROCESSOR_GSUB: &str = "gsub";
+
+const REPLACEMENT_NAME: &str = "replacement";
+const PATTERN_NAME: &str = "pattern";
+
+/// only support string value, and array string value
+#[derive(Debug, Default)]
+pub struct GsubProcessor {
+    fields: Fields,
+    pattern: Option<Regex>,
+    replacement: Option<String>,
+    ignore_missing: bool,
+}
+
+impl GsubProcessor {
+    fn with_fields(&mut self, fields: Fields) {
+        self.fields = fields;
+    }
+
+    fn with_ignore_missing(&mut self, ignore_missing: bool) {
+        self.ignore_missing = ignore_missing;
+    }
+
+    fn try_pattern(&mut self, pattern: &str) -> Result<(), String> {
+        self.pattern = Some(Regex::new(pattern).map_err(|e| e.to_string())?);
+        Ok(())
+    }
+
+    fn with_replacement(&mut self, replacement: impl Into<String>) {
+        self.replacement = Some(replacement.into());
+    }
+
+    fn check(self) -> Result<Self, String> {
+        if self.pattern.is_none() {
+            return Err("pattern is required".to_string());
+        }
+
+        if self.replacement.is_none() {
+            return Err("replacement is required".to_string());
+        }
+
+        Ok(self)
+    }
+
+    fn process_string_field(&self, val: &str, field: &Field) -> Result<Map, String> {
+        let replacement = self.replacement.as_ref().unwrap();
```

Comment:
Do we need to call `check` before `unwrap` the replacement? Or do we have an API contract here?

Reply:
https://github.com/GreptimeTeam/greptimedb/blob/69f5eca96b70a844276b42fb388c9ba36f7953b2/src/pipeline/src/etl/processor/gsub.rs#L147

On constructing the processor, it will check the `replacement` is set in the end. The `Option` here is more for the `default`. But we can use something like `OnceCell` to replace the `Option` in later refactor.